src/main/java/com/digitalasset/testing/ledger/SandboxManager.java

```diff
@@ -142,19 +142,46 @@ private void startSandbox(int port) throws IOException {
     sandboxPort = port;
     sandboxRunner =
         SandboxRunnerFactory.getSandboxRunner(
-            darPath, testModule, testScenario, sandboxPort, useWallclockTime, ledgerId, logLevel);
+            darPath,
+            testModule,
+            testStartScript,
+            sandboxPort,
+            useWallclockTime,
+            ledgerId,
+            logLevel);
     sandboxRunner.startSandbox();
   }
 
-  private void startCommChannels() throws TimeoutException {
+  private void startCommChannels() throws TimeoutException, IOException, InterruptedException {
     channel =
         ManagedChannelBuilder.forAddress("localhost", sandboxPort)
             .usePlaintext()
             .maxInboundMessageSize(Integer.MAX_VALUE)
             .build();
     DamlLedgerClient.Builder builder = DamlLedgerClient.newBuilder("localhost", sandboxPort);
     ledgerClient = builder.build();
-    waitForSandbox(ledgerClient, waitTimeout, logger);
+    try {
+      waitForSandbox(ledgerClient, waitTimeout, logger);
+    } catch (TimeoutException e) {
+      try {
+        sandboxRunner.stopSandbox();
+      } catch (Exception ee) {
+        throw new IOException("Unable to connect to sandbox, and failed to kill it.", ee);
+      }
+      throw e;
+    }
+
+    if (testModule.isPresent() && testStartScript.isPresent()) {
+      DamlScriptRunner scriptRunner =
+          new DamlScriptRunner.Builder()
+              .dar(darPath)
+              .sandboxPort(sandboxPort)
+              .scriptName(String.format("%s:%s", testModule.get(), testStartScript.get()))
+              .useWallclockTime(useWallclockTime)
+              .build();
+      scriptRunner.run();
+    }
```

Comment:
I'd extract it to a method like `runScriptIfConfigured`.

Reply:
okay